src/rcc.rs

```diff
@@ -478,28 +450,276 @@ impl CFGR {
+                self.pllq(q as u8);
+            }
+
+            Some(self)
+        } else {
+            None
+        }
+    }
+
+    /// Configure the default clock settings.
+    /// Set sysclk as 216 Mhz and setup USB clock if defined.
+    pub fn set_defaults(self) -> Self {
+        self.set_sysclk(Hertz(216_000_000));
+
+        self
+    }
+
+    /// Configure the "mandatory" clocks (`sysclk`, `hclk`, `pclk1` and `pclk2')
+    /// and return them via the `Clocks` struct.
+    ///
+    /// The user shouldn't call freeze more than once as the clocks parameters
+    /// cannot be changed after the clocks have started.
+    ///
+    /// The implementation makes the following choice: HSI is always chosen over
+    /// HSE except when HSE is provided. When HSE is provided, HSE is used
+    /// wherever it is possible.
+    pub fn freeze(self) -> Clocks {
+        let flash = unsafe { &(*FLASH::ptr()) };
+        let rcc = unsafe { &(*RCC::ptr()) };
+        let pwr = unsafe { &(*PWR::ptr()) };
+
+        let (clocks, config) = self.calculate_clocks();
+
+        // Switch to fail-safe clock settings.
+        // This is useful when booting from a bootloader that alters clock tree configuration.
+        // Turn on HSI
+        rcc.cr.modify(|_, w| w.hsion().set_bit());
+        while rcc.cr.read().hsirdy().bit_is_clear() {}
+        // Switch to HSI
+        rcc.cfgr.modify(|_, w| w.sw().hsi());
+
+        // Configure HSE if provided
+        if self.hse.is_some() {
+            // Configure the HSE mode
+            match self.hse.as_ref().unwrap().mode {
+                HSEClockMode::Bypass => rcc.cr.modify(|_, w| w.hsebyp().bypassed()),
+                HSEClockMode::Oscillator => rcc.cr.modify(|_, w| w.hsebyp().not_bypassed()),
+            }
+            // Start HSE
+            rcc.cr.modify(|_, w| w.hseon().on());
+            while rcc.cr.read().hserdy().is_not_ready() {}
+        }
+
+        // Enable sequence follows by RM 4.1.4 Entering Overdrive mode.
+        if self.use_pll || self.use_pll48clk {
+            rcc.pllcfgr.write(|w| unsafe {
+                w.pllm().bits(self.pllm);
+                w.plln().bits(self.plln);
+                w.pllp().bits(self.pllp as u8);
+                w.pllq().bits(self.pllq);
+                w.pllsrc().bit(self.hse.is_some())
+            });
+
+            // Enable PWR domain and setup VOSscale and Overdrive options
+            rcc.apb1enr.modify(|_, w| w.pwren().set_bit());
+
+            pwr.cr1.modify(|_, w| match config.vos_scale {
+                VOSscale::PwrScale3 => w.vos().scale3(),
+                VOSscale::PwrScale2 => w.vos().scale2(),
+                VOSscale::PwrScale1 => w.vos().scale1(),
+            });
+
+            if config.overdrive {
+                pwr.cr1.modify(|_, w| w.oden().set_bit());
+            }
+
+            // Enable PLL
+            rcc.cr.modify(|_, w| w.pllon().on());
+
+            // // Wait for PLL to stabilise
```

Comment:
extra `//` here, maybe:  `// Wait for PLL to stabilise`

Reply:
Sure, removed.